src/state/opamRepositoryState.ml

```diff
@@ -214,8 +214,12 @@ let load lock_kind gt =
       then
         let tmp = lazy (
           let tmp_root = Lazy.force repos_tmp_root in
-          OpamFilename.extract_in tar tmp_root;
-          OpamFilename.Op.(tmp_root / OpamRepositoryName.to_string name)
+          if OpamFilename.exists tar then
+            (OpamFilename.extract_in tar tmp_root;
+             OpamFilename.Op.(tmp_root / OpamRepositoryName.to_string name))
+          else
```

Comment:
So IIUC, this will only be triggered if the tar file has been removed since the repo state was loaded (and there was no uncompressed root when it loaded). Where will the `Not_found` be caught ?

It  might be safer to better protect the functions that can remove the tar file (at worst, force a full repo state reload — I agree that this design with hidden state within lazies in the repo state may make correct updates more complicated, esp. with our finalisers... we would need to keep the old binding for cleanup, but not use it anymore when `get_root`/`get_repo_root` is called) ?

Reply:
cleanup is now done on `set_url` call